dhall-csv/src/Dhall/CsvToDhall.hs

```diff
@@ -1,4 +1,24 @@
-module Dhall.CsvToDhall (hello) where
+module Dhall.CsvToDhall (dhallFromCsv) where
 
-hello :: String
-hello = "Hello CSV to Dhall!"
+import Data.Text.Encoding   (decodeUtf8)
+import Data.Void            (Void)
+import Dhall.Core           (Expr, RecordField)
+import Dhall.Src            (Src)
+
+import qualified Data.Csv
+import qualified Data.HashMap.Strict    as HashMap
+import qualified Data.Sequence          as Sequence
+import qualified Dhall.Core             as Core
+import qualified Dhall.Map
+
+dhallFromCsv :: [Data.Csv.NamedRecord] -> Expr Src Void
+dhallFromCsv csv = Core.ListLit Nothing $ Sequence.fromList $ map convertRecord csv
+  where
+    convertRecord :: Data.Csv.NamedRecord -> Expr Src Void
+    convertRecord recordCsv = Core.RecordLit dhallMap
+      where
+        recordTextKeys = HashMap.mapKeys decodeUtf8 recordCsv
+        recordDhallFields = HashMap.map convertField recordTextKeys
+        dhallMap = Dhall.Map.fromList $ HashMap.toList recordDhallFields
```

Comment:
Performance isn't a priority for now, however:

By applying the key and value transformations on the pairs returned by `toList` you could avoid the overhead of hashing the new keys and reconstructing the hash map. `bimap` from `Bifunctor` might be useful for applying the transformations to the tuples.

Reply:
You're right! I didn't know about `Bifunctor`s, it seems cleaner that way thanks